### ISSUE TITLE
Access nil value if file doesn't exist in needMOC

### ISSUE THREAD

**Issue by redorav** [dcourtois/premake-qt#10]
Hi,

I'm using premake-qt in my project and I've found a bug. If I try to add a file that doesn't yet exist to my current project (because it's going to be generated as part of the build process but I want it to appear in the file explorer in Visual Studio) the file goes through the needMOC function, which doesn't check whether a file exists and therefore project generation fails. My suggested fix is to wrap in an if file

```
local file = io.open(filename)
-- scan it to find 'Q_OBJECT' or 'Q_GADGET'
if file then		
    for line in file:lines() do
        if line:find("Q_OBJECT") or line:find("Q_GADGET") then
            needmoc = true
            break
        end
    end
    io.close(file)
end
```

**Comment by dcourtois** [comment]
Hi, you're totally right, thanks for spotting this :)
I'll fix this as soon as I get my internet access back (I just moved, and am currently camping in my new flat without electricity nor internet until next friday :D)

**Comment by redorav** [comment]
That's cool I've got it modified locally until then. Good luck with the move!

**Comment by dcourtois** [comment]
Hi again ! Sorry for the delay : new flat, new job, new lots-of-things and this totally slipped out of my mind :D
Anyway, this issue should be fixed by commit e0b7cf62d1b1091c7c4ad0fe5ff1d891c1fd1f90
It also fixes 2 other potential bugs with non-existing files (one on the .qrc files where the addon checks for dependencies, and the other one in the part which tries to check the Qt version in the SDK)

Thanks again for reporting the issue, and sorry for the long wait :)

**Comment by redorav** [comment]
Cool don't worry about it, I hope the move went well ;) Thanks for looking into it, I'll update as soon as I get home.
